### PR TITLE
bugfix: EG loglevel error for admin and metrics module

### LINKED ISSUE #4339: EG loglevel error for admin and metrics module
*Description*:
>What issue is being seen? Describe what should be happening instead of
the bug, for example: Envoy should not crash, the expected value isn't
returned, etc.

*Repro steps*:
> set the EG logleve to `warn` with envoy-gateway-config but the admin and metrics can not assign the level.

```yaml
apiVersion: v1
data:
  envoy-gateway.yaml: |-
    apiVersion: gateway.envoyproxy.io/v1alpha1
    kind: EnvoyGateway
    logging:
      level:
        default: warn
    gateway:
      controllerName: gateway.envoyproxy.io/gatewayclass-controller
    provider:
      type: Kubernetes
      kubernetes:
        overwriteControlPlaneCerts: true
kind: ConfigMap
```

>**Note**: If there are privacy concerns, sanitize the data prior to
sharing.

*Environment*:
>Include the environment like gateway version, envoy version and so on.

*Logs*:
>Include the access logs and the Envoy logs.
```
2024-09-26T14:43:44.712Z        INFO    admin   admin/server.go:37      starting admin server   {"address": "127.0.0.1:19000", "enablePprof": false}
2024-09-26T14:43:44.712Z        INFO    metrics metrics/register.go:165 initialized metrics pull endpoint       {"address": "0.0.0.0:19001", "endpoint": "/metrics"}
2024-09-26T14:43:44.713Z        INFO    metrics metrics/register.go:54  starting metrics server {"address": "0.0.0.0:19001"}
```

### PR DESCRIPTION
**Pull request by qicz**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #4339

### CODE REVIEW

**Review by zirain — APPROVED** [review]

File: internal/metrics/register.go [hunk 1]
```diff
@@ -42,6 +45,8 @@ func Init(cfg *config.Server) error {
 	}
 
 	if !options.pullOptions.disable {
+		metricsLogger = cfg.Logger.WithName("metrics")
```
Comment (arkodg) [inline internal/metrics/register.go:48]: why not move this to L37 ?

Comment (qicz) [reply internal/metrics/register.go]: done

**Review by zirain — APPROVED** [review]

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Pull request merged**